### PR TITLE
Do not run IHostedService implementations

### LINKED ISSUE #2712: CLI is running all registered IHostedService
Currently the CLI is running all registered Hosted Services (`IHostedService`). I see that it's needed to start the application because otherwise the `app.Use*`/`app.Map*` configuration is not run. Because it's starting the application (with a NoopServer) it is running all registered HostedServices.

Because the CLI is almost always run in the context of a build it would be nice to not run those registered hosted services. This would also solve instances of for example HangFire or other things which start an IHostedService.

I've got a fix which I would like to discuss. It's a bit dirty but it works in my tests.

### PR DESCRIPTION
**Pull request by martincostello**
- Remove `IHostedService` implementations so the CLI doesn't run hosted services (and then hang/fail).
- Refactor CLI tests to reduce duplication.
- Use newer C# syntax as suggested by Visual Studio.

Supersedes #2713.

Resolves #2712.

/cc @desjoerd

### CODE REVIEW

File: src/Swashbuckle.AspNetCore.Cli/HostingApplication.cs [hunk 1]
```diff
@@ -31,6 +33,18 @@ void ConfigureHostBuilder(object hostBuilder)
                 {
                     services.AddSingleton<IServer, NoopServer>();
                     services.AddSingleton<IHostLifetime, NoopHostLifetime>();
+
+                    for (var i = services.Count - 1; i >= 0; i--)
+                    {
+                        // exclude all implementations of IHostedService
+                        // except Microsoft.AspNetCore.Hosting.GenericWebHostService because that one will build/configure
+                        // the WebApplication/Middleware pipeline in the case of the GenericWebHostBuilder.
+                        if (typeof(IHostedService).IsAssignableFrom(services[i].ServiceType)
```
Comment (desjoerd) [inline src/Swashbuckle.AspNetCore.Cli/HostingApplication.cs:42]:
```suggestion
                        if (services[i].ServiceType == typeof(IHostedService)
```

This is taken from https://github.com/RicoSuter/NSwag/pull/4756

**Review by desjoerd — CHANGES_REQUESTED** [review]
I've added the suggestion of the fix I wanted to port :)

File: src/Swashbuckle.AspNetCore.Cli/HostingApplication.cs (hunk 1 above)
Comment (martincostello) [reply src/Swashbuckle.AspNetCore.Cli/HostingApplication.cs:42]: Thanks! Do you have a specific example of a service that shouldn't be removed so I can augment the test?

**Review by mesMess — APPROVED** [review]
Stop running Hosted Service interfaces (some) to reduce hanging
Syntax cleanup and changes (Visual Studio Syntax Guide)

**Pull request merged**